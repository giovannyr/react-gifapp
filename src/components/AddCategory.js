import React, { useState } from "react";
import PropTypes from 'prop-types'

export const AddCategory = ({setCategories}) => {
    const [inputValue, setInputValue] = useState('');

    const handleInputValue = (evt) => {
        setInputValue(evt.target.value);
    };

    const handleSubmit = (evt) => {
        evt.preventDefault();
        if(inputValue.trim().length > 2){
            setCategories( categories => [inputValue, ...categories]);
            setInputValue('');
        }
    };

    return (
        <form onSubmit={handleSubmit}>
            <input type="text" value={inputValue} onChange={handleInputValue} />
        </form>
    );
};

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
