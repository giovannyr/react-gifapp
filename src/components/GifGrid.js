import React from "react";
import PropTypes from "prop-types";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";

export const GifGrid = ({ category }) => {
    const { data: images, loading } = useFetchGifs(category);

    return (
        <>
            <h3 className="animate__animated animate__flipInX animate__slow">
                {category}
            </h3>

            {loading && (
                <p className="animate__animated animate__flash animate__delay-2s">
                    cargando...
                </p>
            )}

            <div className="card-grid">
                {images.map((image) => (
                    <GifGridItem key={image.id} {...image} />
                ))}
            </div>
        </>
    );
};

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
};
