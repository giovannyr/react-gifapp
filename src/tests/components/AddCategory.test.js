import "@testing-library/jest-dom";
import React from "react";
import { shallow } from "enzyme";
import { AddCategory } from "../../components/AddCategory";

describe("pruebas <AddCategory />", () => {
    const setCategories = jest.fn();
    let wrapper;

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories} />);
    });

    test("debe mostrar componente correctamente ", () => {
        expect(wrapper).toMatchSnapshot();
    });

    test("debe simular evento change", () => {
        const input = wrapper.find("input");
        const value = "Hola mundo";

        input.simulate("change", {
            target: { value: value },
        });
    });

    test("No debe postear la informacion con submit", () => {
        // wrapper.find("form").simulate("submit", { preventDefault: () => {} });
        // FORMA ABREVIADA
        wrapper.find("form").simulate("submit", { preventDefault() {} });

        expect(setCategories).not.toHaveBeenCalled();
    });

    test("debe llamar setCategories y limpiart caja de texto", () => {
        const value = "Hola mundo";

        // 1. simular input change
        wrapper.find("input").simulate("change", { target: { value } });

        // 2. simular submit
        wrapper.find("form").simulate("submit", { preventDefault() {} });

        // 3. setCategories se debe haber llamado
        expect(setCategories).toHaveBeenCalled();
        expect(setCategories).toHaveBeenCalledTimes(1);
        expect(setCategories).toHaveBeenCalledWith(expect.any(Function));

        // 4. el valor del imput debe estar ''
        expect(wrapper.find("input").prop("value")).toBe("");
    });
});
