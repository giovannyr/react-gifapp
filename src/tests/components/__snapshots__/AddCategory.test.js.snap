// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas <AddCategory /> debe mostrar componente correctamente  1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
